javascript/listview.js

```diff
@@ -1,81 +1,200 @@
-divTitleAndDistance.setAttribute("class", "title-and-distancefrom");
-infoSectionSelector.appendChild(divTitleAndDistance);
-
-var divReviews = document.createElement('div');
-divReviews.setAttribute("class", "reviews");
-infoSectionSelector.appendChild(divReviews);
-
-var divAddress = document.createElement('div');
-divAddress.setAttribute("class", "address");
-infoSectionSelector.appendChild(divAddress);
-
-var divRouteTimeAndLength = document.createElement('div');
-divRouteTimeAndLength.setAttribute("class", "route-time-and-length");
-infoSectionSelector.appendChild(divRouteTimeAndLength);
-
-var divContributer = document.createElement('div');
-divContributer.setAttribute("class", "contributer");
-infoSectionSelector.appendChild(divContributer);
-
-// Info Section - Title and Distance Row
-var titleAndDistanceSelector = document.querySelector('.title-and-distancefrom');
-var divRouteTitle = document.createElement('div');
-divRouteTitle.setAttribute("class", "route-title");
-divRouteTitle.textContent = 'Graffiti'
-titleAndDistanceSelector.appendChild(divRouteTitle);
-
-var divDistanceFrom = document.createElement('div');
-divDistanceFrom.setAttribute("class", "distancefrom");
-var divDistanceIcon = document.createElement('div');
-divDistanceIcon.setAttribute("class", "distancefrom-icon");
-var divDistanceNumber = document.createElement('div');
-divDistanceNumber.setAttribute("class", "distancefrom-number");
-divDistanceNumber.textContent = '1.2 mi'
-var imgDistanceIcon = document.createElement('img');
-imgDistanceIcon.setAttribute("class", "nearme");
-imgDistanceIcon.setAttribute("src", "images/nearme.png");
-
-titleAndDistanceSelector.appendChild(divDistanceFrom);
-divDistanceFrom.appendChild(divDistanceIcon);
-divDistanceFrom.appendChild(divDistanceNumber);
-divDistanceIcon.appendChild(imgDistanceIcon);
-
-// Info Section - Review Row
-
-
+var getTitleValue = function(title) {
+    var dbTitleRef = firebase.database().ref('routes/-L7Aa3dtKeXFFX8SLOTF/title');
+    dbTitleRef.on('value', snap => title.textContent = snap.val());
+};
+
+var getDistanceFromValue = function(selector) {
+    // This function will need to calculate distance
+    // between tour and user location in the future
+    var calculateDistance = '1.2 mi';
+    selector.textContent = calculateDistance;
+};
+
+var getAmountOfStars = function() {
+    // Creates number of stars based on overall rating
+    // Blocked on getting value out
+
+    // var dbRatingRef = firebase.database().ref('routes/-L7Aa3dtKeXFFX8SLOTF/rating');
+    // dbRatingRef.on('value', snap => reviewsSelector.textContent = snap.val());
+
+};
+
+var getNumberOfReviewsValue = function(reviews) {
+    // Filled with placeholder values
+    var dbReviewsRef = firebase.database().ref('routes/-L7Aa3dtKeXFFX8SLOTF/reviews');
+    dbReviewsRef.on('value', snap => reviews.textContent = snap.val() + ' Reviews');
+};
+
+var getAddressValue = function(address) {
```

Comment:
rather than getting every attribute separately from the database, it is better to import the whole object and then get the attributes you need.  Making so many call make actually push us up against are API call limit theoretically.

`var fireBaseRef = firebase.database().ref('routes');
var objectId = '-L7LE7Sokvs9dj9mTGmg';
var fireBaseObject = firebase.database().ref('routes/' + objectId);
fireBaseObject.on("value", function(snapshot) {
var dbAddressRef = snapshot.value['address'];
var dbCityRef = snapshot.value['city'];
var dbStateRef = snapshot.value['state'];
`

Reply:
with the refactor, lines 33-56 can be under one call.